src/frontend/src/expr/mod.rs

```diff
@@ -291,6 +291,30 @@ macro_rules! impl_has_variant {
 
 impl_has_variant! {InputRef, Literal, FunctionCall, AggCall, Subquery, TableFunction, WindowFunction}
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) struct InequalityInputPair {
```

Comment:
Could you rename it to `BandJoinCondition` too? Also in other places. the "Inequality" sounds too vague